src/interface/src/app/treatments/direct-impacts/metric-filters/metric-filters.component.ts

```diff
@@ -0,0 +1,158 @@
+  {
+    id: 'ID_MORTALITY',
+    color: '',
+    label: 'Mortality',
+  },
+  {
+    id: 'ID_POTENTIAL_SMOKE',
+    color: '',
+    label: 'Potential Smoke',
+  },
+  {
+    id: 'ID_PROBABILITY_OF_TORCHING',
+    color: '',
+    label: 'Probability of Torching',
+  },
+  {
+    id: 'ID_QUADRATIC_MEAN_DIAMETER',
+    color: '',
+    label: 'Quadratic Mean Diameter',
+  },
+  {
+    id: 'ID_STAND_DENSITY_INDEX',
+    color: '',
+    label: 'Stand Density Index',
+  },
+  {
+    id: 'ID_TOTAL_HEIGHT',
+    color: '',
+    label: 'Total Height',
+  },
+  {
+    id: 'ID_TOTAL_FLAME_SEVERITY',
+    color: '',
+    label: 'Total Flame Severity',
+  },
+  {
+    id: 'ID_TOTAL_CARBON',
+    color: '',
+    label: 'Total Carbon',
+  },
+];
+
+@Component({
+  selector: 'app-metric-filters',
+  standalone: true,
+  imports: [NgFor, MetricSelectorComponent, MetricSelectorComponent],
+  templateUrl: './metric-filters.component.html',
+  styleUrl: './metric-filters.component.scss',
+})
+export class MetricFiltersComponent implements OnInit {
+  @Output() metricSelected = new EventEmitter<Metric>();
+
+  initialOptions: Metric[] = METRICS;
+
+  metricColors: string[] = ['#4361EE', '#9071E8', '#EC933A', '#63C2A2'];
+
+  // Initializing every dropdown
+  dropdownOptions: Metric[][] = [
+    [...this.initialOptions],
+    [...this.initialOptions],
+    [...this.initialOptions],
+    [...this.initialOptions],
+  ];
+
+  // Storing the selected IDs
+  selectedOptions: string[] = [
+    this.initialOptions[0].id,
+    this.initialOptions[1].id,
+    this.initialOptions[2].id,
+    this.initialOptions[3].id,
+  ];
+
+  ngOnInit(): void {
+    // Updating every list based on the default selected values
+    this.updateDropdownOptions(null);
+  }
+
+  optionSelected(dropdownIndex: number, metric: Metric): void {
+    // Storing the selection
+    this.selectedOptions[dropdownIndex] = metric.id;
+
+    // Updating the dropdowns
+    this.updateDropdownOptions(dropdownIndex);
+  }
+
+  updateDropdownOptions(updatedDropdownIndex: number | null): void {
+    // Updating every dropdown with the available options
+    this.dropdownOptions = this.dropdownOptions.map(
+      (list: Metric[], listIndex: number) => {
+        // if the list is the same as the dropdown we are updating we want to keep it as it is
+        if (listIndex === updatedDropdownIndex) {
+          return this.applyMetricColor(listIndex, list);
+        }
+
+        // Keeping the selected option if the dropdown already have value and is not the updated dropdown
+        const filteredList = this.initialOptions.filter(
+          (m) =>
+            !this.selectedOptions.includes(m.id) ||
+            m.id === this.selectedOptions[listIndex]
+        );
+
+        return this.applyMetricColor(listIndex, filteredList);
+      }
+    );
+  }
+
+  applyMetricColor(dropdownIndex: number, list: Metric[]): Metric[] {
+    return list.map((metric) => {
+      metric.color = this.metricColors[dropdownIndex];
```

Comment:
This is the part i think is a bit strange. We are updating the color on the metric, but what we should be actually be doing is updating the metric on the given slot (1, 2, 3 or 4).
The color on each slot is static.
In other words, if you let the "slot" handle the color itself, you might not need this method at all

Reply:
You were right, I updated it and removed that method